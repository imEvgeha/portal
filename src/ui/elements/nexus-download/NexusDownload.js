import React from 'react';
import PropTypes from 'prop-types';
import Button from '@atlaskit/button';
import {download} from '../../../util/Common';
import {DOWNLOAD, JSON_MIME} from './constants';

const NexusDownload = ({data, filename, mimeType, label}) => {
    const handleDownload = () => {
        download(JSON.stringify(data), filename, mimeType);
    };

    return (
        <Button onClick={handleDownload}>{label}</Button>
    );
};

NexusDownload.propTypes = {
    data: PropTypes.object.isRequired,
    filename: PropTypes.string.isRequired,
    mimeType: PropTypes.string,
    label: PropTypes.string,
};

NexusDownload.defaultProps = {
    label: DOWNLOAD,
    mimeType: JSON_MIME,
};

export default NexusDownload;
