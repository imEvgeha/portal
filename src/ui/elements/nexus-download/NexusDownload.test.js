import React from 'react';
import {shallow} from 'enzyme';
import FileSaver from 'file-saver';
import NexusDownload from './NexusDownload';

describe('NexusDownload', () => {
    let wrapper;
    const props = {
        data: { test: 'awesome string'},
        filename: 'event_filename',
        mimeType: 'application/json',
    };

    global.Blob = (content, options) => ({content, options});
    global.URL.createObjectURL = jest.fn();

    beforeEach(() => {
        wrapper = shallow(<NexusDownload {...props} />);
    });

    it('should render the component', () => {
        expect(wrapper).not.toBe(null);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call file-saver with proper arguments', () => {
        const button = wrapper.find('.nexus-c-download');
        expect(button.length).toEqual(1);

        const spy = jest.spyOn(FileSaver, 'saveAs');

        button.simulate('click');

        expect(spy).toHaveBeenCalledWith({
            content: [JSON.stringify(props.data)],
            options: {
                type: props.mimeType
            }
        }, props.filename);

        spy.mockRestore();
    });
});
