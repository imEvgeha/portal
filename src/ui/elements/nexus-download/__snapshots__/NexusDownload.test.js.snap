// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NexusDownload should match snapshot 1`] = `
<WithAnalyticsContext(WithAnalyticsEvents(Button))
  className="nexus-c-download"
  onClick={[Function]}
>
  Download
</WithAnalyticsContext(WithAnalyticsEvents(Button))>
`;
